docs/src/main/sphinx/connector/faker.md

```diff
@@ -0,0 +1,225 @@
+# Faker connector
+
+The Faker connector generates random data matching a defined structure. It uses
+the [Datafaker](https://www.datafaker.net/) library, which makes the generated
+data more realistic. It can be used to easily populate another data source with
+large and realistic test data, which can help with testing performance of
+applications processing data (like Trino itself), and also application's user
+interface.
+
+## Usage
+
+Faker generates data when reading from a table created in a catalog using this
+connector. This makes it easy to fill an existing schema with random data, by
+copying only the schema into a Faker catalog, and inserting the data back into
+the original tables.
+
+Create a catalog properties file that specifies the Faker connector by setting
+the `connector.name` to `faker`.
+
+For example, to access data as the `generator` catalog, create the file
+`etc/catalog/generator.properties`. Replace the connection properties as
+appropriate for your setup:
+
+
+```text
+connector.name=faker
+faker.null-probability=0.1
+faker.default-limit=1000
+```
+
+Create a table with the same columns as in the table to be populated with
+random data. Exclude all properties, because the Faker connector doesn't
+support same properties as other connectors.
+
+```sql
+CREATE TABLE generator.default.customer (LIKE production.public.customer EXCLUDING PROPERTIES);
+```
+
+Insert random data into the original table, by selecting it from the Faker
+connector. Data generated for columns of non-character types cover the whole
+range of that data type, so to make it more practical, adding constraints might
+be necessary.
+
+```sql
+INSERT INTO production.public.customers
+SELECT *
+FROM generator.default.customers
+WHERE
+  born_at BETWEEN CURRENT_DATE - INTERVAL '150' YEAR AND CURRENT_DATE
+  AND age_years BETWEEN 0 AND 150
+LIMIT 100;
+```
+
+To generate even more realistic data, choose specific generators by setting the
+`generator` property on columns. Start with getting the complete definition of a table.
+
+```sql
+SHOW CREATE TABLE production.public.customers;
+```
+
+Modify the output of the previous query and add some column properties.
+
+```sql
+CREATE TABLE generator.default.customer (
+  id UUID NOT NULL,
+  name VARCHAR NOT NULL WITH (generator = '#{Name.first_name} #{Name.last_name}'),
```

Comment:
So somewhere we have to document this table property .. probably in the Configuration section .. so far this is only implied but not explicitly documented

Reply:
I added tables for all schema, table and column properties.